modules/microshift-cni.adoc

```diff
@@ -0,0 +1,301 @@
+A wide range of networking features are available with OVN-Kubernetes, such as:
+
+* Kubernetes network policy
+* Dynamic node IP
+* Cluster network on specified host interface
+* Secondary gateway interface
+* Dual stack
+* Service idling (disabled)
+* Egress firewall/QOS/IP (disabled)
+* Cluster network on specified host interface
+//these don't match the default values in the OCP docs; what features are available/relevant to MicroShift
+//https://docs.openshift.com/container-platform/4.11/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html#nw-ovn-kubernetes-matrix_about-ovn-kubernetes
+
+== {product-title} networking components overview
+
+This brief overview describes networking components and their operation in {product-title}.
+//Q: are there immutable settings we should tell users about?
+NetworkManager::
+NetworkManager is required by OVN-Kubernetes to set up the initial gateway bridge on the {product-title} node. NetworkManager and NetworkManager-OVS RPM packages are installed as dependencies to the `microshift-networking` RPM package. NetworkManager is configured to use the `keyfile` plugin by default and restarts after installation of the `microshift-networking` RPM package.
+//Q: Meaning the microshift networking RPM replaces the keyfile plugin?
+microshift-ovs-init::
+The `microshift-ovs-init.service` is installed by the `microshift-networking` RPM package as a one-shot systemd service. The 'microshift-ovs-init.service` executes the `configure-ovs.sh script` which uses NetworkManager commands to set up the OVS gateway bridge.
+
+OVN containers::
+OVN-Kubernetes cluster manifests are in the `microshift/assets/components/ovn` directory. Two OVN-Kubernetes daemon sets are rendered and applied by {product-title}.
+
+ovnkube-master::
+Includes the `northd`, `nbdb`, `sbdb` and `ovnkube-master` containers.
+
+ovnkube-node::
+The ovnkube-node includes the OVN-Controller container. After {product-title} boots, the OVN-Kubernetes daemon sets are deployed in the `openshift-ovn-kubernetes` namespace.
+
+Packaging::
+OVN-Kubernetes manifests and startup logic are built into the {product-title} RPM. The systemd services and configurations included in `microshift-networking` RPM are:
+
+* microshift-nm.conf for NetworkManager.service
+* microshift-cpuaffinity.conf for ovs-vswitchd.service
+* microshift-cpuaffinity.conf for ovsdb-server.service
+* microshift-ovs-init.service
+* configure-ovs.sh for microshift-ovs-init.service
+* configure-ovs-microshift.sh for microshift-ovs-init.service
+
+== Configuring OVN-Kubernetes
+An OVN-Kubernetes config file must be written to /etc/microshift/ovn.yaml. {product-title} will use default OVN-Kubernetes configuration values if an OVN-Kubernetes config file is not customized.
+//Q: what are the default values?
+
+.Supported OVN-Kubernetes configurations for {product-title}.
+[cols="6",options="header"]
+|===
+|Field
+|Required
+|Type
+|Default
+|Description
+|Example
+
+|`disableOVSInit`
```

Comment:
now there is also gatewayInterface
and externalGatewayInterface

and there are two levels, we just made your table more complicated :(

Reply:
Hmm, I don't understand how this should look now.